core/src/main/java/org/mule/runtime/core/internal/security/DefaultMuleSecurityManager.java

```diff
@@ -80,6 +80,14 @@ public Authentication authenticate(Authentication authentication) throws Securit
         Authentication result = null;
         try {
           result = provider.authenticate(authentication);
+        } catch (UnauthorisedException e) {
```

Comment:
this `catch` can be removed since it does the same as the `catch Exception` at the bottom

Reply:
The problem is that `UnauthorisedException` is a subtype of `SecurityException`, so in that case it will get into that `catch` instead of the `catch (Exception e)` one

Reply:
An alternative could be leaving just the SecurityException catch, but adding there:
``` 
catch (SecurityException e) {
    if (!iter.hasNext()) {
        if (e instanceof UnauthorisedException) {
            throw new UnauthorisedException(authorizationAttemptFailed(), e);
        } else {
            throw new SecurityException(authorizationAttemptFailed(), e);
        }
    }
} ...
```